src/test/java/com/es/phoneshop/service/implementation/ProductServiceImplTest.java

```diff
@@ -0,0 +1,161 @@
+package com.es.phoneshop.service.implementation;
+
+import com.es.phoneshop.dao.ProductDao;
+import com.es.phoneshop.dao.implementation.ArrayListProductDaoImpl;
+import com.es.phoneshop.exception.ProductNotFoundException;
+import com.es.phoneshop.model.Product;
+import com.es.phoneshop.service.ProductService;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+
+import java.math.BigDecimal;
+import java.util.Currency;
+import java.util.List;
+import java.util.UUID;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class ProductServiceImplTest {
+    private ProductService productService;
+    private Currency usd;
+
+    @Mock
+    private ProductDao productDao;
+
+    @Before
+    public void setup() {
+        usd = Currency.getInstance("USD");
+        productService = new ProductServiceImpl();
+        productDao = new ArrayListProductDaoImpl();
+    }
+
+    @Test
+    public void getProduct_ValidId_ShouldReturnProductWithIdFromDao() {
+        // given
+        final UUID id = productService.findProducts().get(0).getId();
+        final BigDecimal expectedPrice = new BigDecimal(100);
+        final String expectedDescription = "Samsung Galaxy S";
+        final int expectedStock = 100;
+
+        // when
+        Product product = productService.getProduct(id);
+
+        // then
+        assertNotNull(product);
+        assertEquals(expectedPrice, product.getPrice());
+        assertEquals(expectedDescription, product.getDescription());
+        assertEquals(expectedStock, product.getStock());
+    }
+
+    @Test(expected = ProductNotFoundException.class)
+    public void getProduct_InvalidId_ShouldThrowProductNotFoundException() {
+        // given
+        final UUID id = UUID.randomUUID();
+
+        // when
+        productService.getProduct(id);
+
+        // then (exception is thrown)
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void getProduct_NullId_ShouldThrowIllegalArgumentException() {
+        // when
+        productService.getProduct(null);
+
+        // then (exception is thrown)
+    }
+
+    @Test
+    public void findProducts_ShouldReturnProductListFromDao() {
+        // when
+        List<Product> products = productService.findProducts();
+
+        // then
+        assertFalse(products.isEmpty());
+    }
+
+    @Test
+    public void findProducts_ProductWithZeroStock_ShouldNotIncludeProductToList() {
+        // given
+        final int stock = 0;
+        final Product productWithZeroStock = new Product("simsxg75", "Siemens SXG75", new BigDecimal(150), usd, stock, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Siemens/Siemens%20SXG75.jpg");
+        productDao.save(productWithZeroStock);
```

Comment:
you saved product into another instance of productDao that isn't related to tested productService